Fixes for issues with adding/removing OC's and Season3

AnthonyMichaelTDM opened the pull request:
### What this PR does:
 - fix the issues identified/references in issues #37 , #40 , #41 , and #42
 - updated [readme.md](https://github.com/AnthonyMichaelTDM/DRG-Save-Editor/blob/master/readme.md)
 - add the OCs for the new secondaries (#46)
 - add editor.ui to the base directory so that people can run the project with python, and instructions on how to do so
 - support for non-windows operating systems
 - updates for season 3 (#53)
 
### what this PR doesn't do:
 - rebuild [DRG Save Editor.zip](https://github.com/robertnunn/DRG-Save-Editor/blob/master/DRG%20Save%20Editor.zip)

### notes and acknowledgements 

special thanks to @wahlp , who's responses to issue 37 pointed me in the right direction
special thanks to @JairRmz for supplying 20 missing OC GUIDs 

for people who want to use this working fork, I direct your attention to, well, [my fork](https://github.com/AnthonyMichaelTDM/DRG-Save-Editor)

AnthonyMichaelTDM committed:
added test saves

AnthonyMichaelTDM committed:
.

AnthonyMichaelTDM committed:
.

AnthonyMichaelTDM committed:
normalized save files?

AnthonyMichaelTDM committed:
test

AnthonyMichaelTDM committed:
Merge branch 'master' of github.com:AnthonyMichaelTDM/DRG-Save-Editor

AnthonyMichaelTDM committed:
.

AnthonyMichaelTDM committed:
Merge branch 'master' of github.com:AnthonyMichaelTDM/DRG-Save-Editor

hi

AnthonyMichaelTDM committed:
added another example save file

AnthonyMichaelTDM committed:
reorganize example save files

AnthonyMichaelTDM committed:
re-enable functionality for adding unforged OC's

AnthonyMichaelTDM committed:
add editor.ui to main directory

AnthonyMichaelTDM committed:
fix dumb error

AnthonyMichaelTDM committed:
was using wrong size integer to write some values

AnthonyMichaelTDM committed:
remove test saves bc now they're bloat

AnthonyMichaelTDM commented:
the reason this PR doesn't rebuild the .zip file is bc the build software this project was built around requires you to purchase a pro license in order to use, which is a problem bc I'm broke

AnthonyMichaelTDM committed:
update readme to reflect changes

AnthonyMichaelTDM committed:
Update readme.md

AnthonyMichaelTDM reviewed (commented):
0.0

JairRmz commented:
how can I test this version or any way someone could add it to the ZIP file even manually?

AnthonyMichaelTDM commented:
follow the instructions i added to the readme

basically, clone my fork of the repo, install the required packages w/ pip, and run the project with python

JairRmz commented:
ok, from what I tested with my own save adding overclocks works, yet there are still missing some secondaries as pointed in another error for engineer, driller and scout, how could I help on detecting the overclocks?

AnthonyMichaelTDM commented:
> ok, from what I tested with my own save adding overclocks works, yet there are still missing some secondaries as pointed in another error for engineer, driller and scout, how could I help on detecting the overclocks?

honestly, I'm still trying to figure that out, but if you have a save with some of the new OCs unforged, send it my way in #46

JairRmz commented:
currently crafting every overclock I didnt have, will ignore the OC that arent on the list

AnthonyMichaelTDM commented:
thanks

AnthonyMichaelTDM committed:
add Boltsharks "Fire Bold" & "The Specialist" OCs

AnthonyMichaelTDM committed:
Merge branch 'master' of https://github.com/AnthonyMichaelTDM/DRG-Save-Editor

AnthonyMichaelTDM committed:
add coil guns Backfeeding Module

AnthonyMichaelTDM committed:
4 more OC's added

AnthonyMichaelTDM committed:
9 more OCs added

AnthonyMichaelTDM committed:
remaining 7 ocs added

Sclhum commented:
I have no idea how to do the steps you've provided. I don't know how to open a terminal. I don't know how to run a python command. I have no idea how to do anything. Please help.

AnthonyMichaelTDM commented:
https://www.google.com/search?q=how+to+use+python

Sclhum commented:
I have python downloaded, but I don't know how to copy a "fork," or execute the commands listed. I have github downloaded as well, if that helps.

Sclhum commented:
Attempts to use "pip" result in invalid syntax?

JoshuaAng10 commented:
> Attempts to use "pip" result in invalid syntax?

you need to download powershell, and run the powershell terminal from the file. Then follow the instructions and it should work. Also, make sure python has pip installed. If u still have questions, drop me a DM on discord Joshua#3257, will see if I can help you

Sclhum commented:
I am sorry for being stupid, I figured it out. Works perfectly, A+ script.

Omega172 commented:
[Save.txt](https://github.com/robertnunn/DRG-Save-Editor/files/9078468/Save.txt)
Here is a save with only some of the cosmetic forge items

duarm committed:
linux fix

Sclhum commented:
I know there's a way to change the hexadecimal of other cosmetics to change them into other cosmetics (I changed to get the Head Honcho hat), maybe there's a way you can merge that method with this save editor? 
(Context: https://www.youtube.com/watch?v=9sBAuFtL1x8)

JairRmz commented:
> I know there's a way to change the hexadecimal of other cosmetics to change them into other cosmetics (I changed to get the Head Honcho hat), maybe there's a way you can merge that method with this save editor? (Context: https://www.youtube.com/watch?v=9sBAuFtL1x8)

tbf that method doesnt work without already having a skin/cosmetic that you can change the guid value to, therefore it wouldnt be recommended since creating an instance on the .exe that selects a specific item to edit would make it tiresome, I have tried adding skins before but it wasnt working due to how the items are saved, which kinda follow the same issue with overclocks from what I had tested

AnthonyMichaelTDM commented:
> I know there's a way to change the hexadecimal of other cosmetics to change them into other cosmetics (I changed to get the Head Honcho hat), maybe there's a way you can merge that method with this save editor? (Context: https://www.youtube.com/watch?v=9sBAuFtL1x8)

as @JairRmz said, it would be possible, but due to the shear quantity of ingame cosmetics it would be a very difficult endeavor for very little gain, and probably outside of the scope of this project

RapanMasta commented:
here is a save with different cosmetics and overclocking
[ID_Player.zip](https://github.com/robertnunn/DRG-Save-Editor/files/9214338/ID_Player.zip)

AnthonyMichaelTDM committed:
Merge pull request #1 from duarm/master

do not import winreg on linux

AnthonyMichaelTDM committed:
Update readme.md

AnthonyMichaelTDM committed:
Update readme.md

AnthonyMichaelTDM committed:
Update readme.md

I just figured out that I can change a setting to allow people to open issues on this repo, updated readme to reflect that

AnthonyMichaelTDM committed:
formatting issue in readme fixed

Kit-Harder commented:
Hello.
I seem to have a trouble understanding with the instructions, a more detailed explanation on it would do. 

I did everything I could on trying to follow the instructions but with no progress
![image_2022-07-31_222303377](https://user-images.githubusercontent.com/103932508/182030848-4e63ac6a-9988-4f63-a166-cc7e12bfaa8c.png)
I downloaded the source code but I'm clueless on the second part on how to do that

I am aware that I need python in order for it to work but I tried searching how to use python but I spent 5 hours watching and reading on it yet I still have no idea what im doing

Can anyone help me with this? I know its too much to ask but I just want to know if i can add Overclocks to my save files or not

AnthonyMichaelTDM commented:
You could open an issue on my repository, but a simple google search gives you this [https://www.groovypost.com/howto/open-command-window-terminal-window-specific-folder-windows-mac-linux/](https://www.groovypost.com/howto/open-command-window-terminal-window-specific-folder-windows-mac-linux/)

If you're on windows, you're going to want to open command prompt maybe(?), and you can get python and pip from the windows store I think
Another simple google search of "how to run a python project" yields this: [https://stackoverflow.com/questions/12873542/how-to-run-a-python-project](https://stackoverflow.com/questions/12873542/how-to-run-a-python-project)

Kit-Harder commented:
> You could open an issue on my repository, but a simple google search gives you this https://www.groovypost.com/howto/open-command-window-terminal-window-specific-folder-windows-mac-linux/
> 
> If you're on windows, you're going to want to open command prompt maybe(?), and you can get python and pip from the windows store I think Another simple google search of "how to run a python project" yields this: https://stackoverflow.com/questions/12873542/how-to-run-a-python-project

I finally got it to work, Thank so much!

AnthonyMichaelTDM committed:
fix formatting

AnthonyMichaelTDM committed:
Update readme.md

AnthonyMichaelTDM committed:
Create pylint.yml

AnthonyMichaelTDM committed:
Delete .github/workflows directory

ajm0208 commented:
I get through everything until the last step where I input the command and it says "Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases." I have already tried doing the paths solution, the setting solution, restarting my computer, reinstalling python, and none of them seem to work. Every time I end up with the same error. Any one have any ideas?

AnthonyMichaelTDM commented:
I suggest following the instructions and running the python command without arguments

ajm0208 commented:
> I suggest following the instructions and running the python command without arguments

I don't know how to do that. Searching it up just brings up the things I already tried.

AnthonyMichaelTDM committed:
fixes for season 3

AnthonyMichaelTDM commented:
updated for season 3

AnthonyMichaelTDM committed:
Update readme.md

RisingArmechadon commented:
> currently crafting every overclock I didnt have, will ignore the OC that arent on the list

How do I add overclocks? It doesnt work for me

amogusenjoyee commented:
I tried to add my overclocks to the secondary but it appears as forged and it doesn't show up in the equipment section, how do i fix it?
